test/src/unittests/tonal/test_pitchcontours.py

```diff
@@ -0,0 +1,138 @@
+        emptyPeakBins = [[],[]]
+        emptyPeakSaliences = [[],[]]
+        theHopSize= 2*defaultHopSize
+        bins, saliences, startTimes, duration = PitchContours(hopSize=theHopSize)(emptyPeakBins, emptyPeakSaliences)       
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.        
+        calculatedDuration = (2*theHopSize)/defaultSampleRate
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+    def testZeros(self):
+        theHopSize = 4*defaultHopSize
+        bins, saliences, startTimes, duration = PitchContours(hopSize=theHopSize)(array(zeros([2,256])), array(zeros([2,256])))      
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.
+        calculatedDuration = (2*theHopSize)/defaultSampleRate            
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+    #256 frames x single zero-salience peak at the same cent bin in each frame.
+    def testSingleZeroSaliencePeak(self):
+        peakBins = array(zeros([1, 256]))
+        peakSaliences = array(zeros([1, 256]))
+
+        theHopSize = 4*defaultHopSize
+        bins, saliences, startTimes, duration = PitchContours(hopSize=theHopSize)(peakBins, peakSaliences)      
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.
+        calculatedDuration = (2*theHopSize)/defaultSampleRate            
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+    #256 frames x small random number of zero-salience peaks at random cent bins in each frame (use a fixed random seed).
+    def testVariousZeroSaliencePeaks(self):
+        random.seed(10)
+        n = random.randrange(2, 16, 1)    
+        peakBins = array(zeros([n,256]))
+        peakSaliences = array(zeros([n,256]))
+
+        theHopSize = 4*defaultHopSize
+        bins, saliences, startTimes, duration = PitchContours(hopSize=theHopSize)(array(zeros([2,256])), array(zeros([2,256])))      
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+
+        calculatedDuration = (2*theHopSize)/defaultSampleRate            
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+    def testUnequalInputs(self):        
+        peakBins = [zeros(4096), zeros(4096)]
+        peakSaliences = [zeros(1024), zeros(1024)]
+        self.assertRaises(RuntimeError, lambda: PitchContours()(peakBins, peakSaliences))
+    
+suite = allTests(TestPitchContours)
```

Comment:
The pitch ladder tests make sense for this particular algorithm instead of PitchSalienceFunctionPeaks.

Reply:
testRegressionSynthetic is added.
The synthetic outputs of  contours_bins, contours_saliences, contours_start_times are saved in NPY files

Reply:
added